service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -695,6 +708,82 @@ public ResponseEntity<ApiRegions> listValidRegions(UUID workspaceId, ApiCloudPla
     return new ResponseEntity<>(apiRegions, HttpStatus.OK);
   }
 
+  @Override
+  public ResponseEntity<ApiWsmPolicyExplainResult> explainPolicies(
+      UUID workspaceId, Integer depth) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    workspaceService.validateWorkspaceAndAction(userRequest, workspaceId, SamWorkspaceAction.READ);
+    TpsPaoExplainResult explainResult = tpsApiDispatch.explain(workspaceId, depth);
```

Comment:
Hmm... not so sure about this.
I think we should do the TPS work and related lookups in WorkspaceService. Not in the controller.
I see the advantage of only converting once from TPS form to WSM API form. But I think it would be good hygiene to have an internal WSM model that we build and populate in the service and then do a trivial conversion here in the controller.

Reply:
done